process_report/process_report.py

```diff
@@ -386,6 +399,54 @@ def export_HU_only(dataframe, output_file):
     HU_projects.to_csv(output_file)
 
 
+def export_BU_only(dataframe: pandas.DataFrame, output_file, subsidy_amount):
+    def get_project(row):
+        project_alloc = row[PROJECT_FIELD]
+        if project_alloc.rfind("-") == -1:
+            return project_alloc
+        else:
+            return project_alloc[: project_alloc.rfind("-")]
+
+    BU_projects = dataframe[dataframe[INSTITUTION_FIELD] == "Boston University"]
+    BU_projects["Project"] = BU_projects.apply(get_project, axis=1)
+    BU_projects[SUBSIDY_FIELD] = 0
+    BU_projects = BU_projects[
+        [
+            INVOICE_DATE_FIELD,
+            PI_FIELD,
+            "Project",
+            COST_FIELD,
+            CREDIT_FIELD,
+            SUBSIDY_FIELD,
+            BALANCE_FIELD,
+        ]
+    ]
+    BU_projects = _apply_subsidy(BU_projects, subsidy_amount)
+    BU_projects.to_csv(output_file)
+
+
+def _apply_subsidy(dataframe, subsidy_amount):
+    pi_list = dataframe[PI_FIELD].unique()
+
+    for pi in pi_list:
+        pi_projects = dataframe[dataframe[PI_FIELD] == pi]
+        remaining_subsidy = subsidy_amount
+        for i, row in pi_projects.iterrows():
+            project_remaining_cost = row[
+                BALANCE_FIELD
+            ]  # Applying BU Subsidy after applying credits, so Balance shows rem costs
```

Comment:
This function doesn't know (nor should it know) when it's being called therefore this comment is wrong. The BU Subsidy is applied after credits *because* of when the `export_BU_only` is called.